compiler/il/OMROpcodes.hpp

```diff
@@ -0,0 +1,11786 @@
+      TR::BadILOp, /* .booleanCompareOpCode */ \
+      TR::BadILOp, /* .ifCompareOpCode */ \
+      lconst, /* load long integer constant (64-bit signed 2's complement) */ \
+   ) \
+   MACRO(\
+      TR::fconst, /* .opcode */ \
+      "fconst", /* .name */ \
+      ILProp1::LoadConst, /* .properties1 */ \
+      ILProp2::ValueNumberShare, /* .properties2 */ \
+      ILProp3::LikeUse, /* .properties3 */ \
+      0, /* .properties4 */ \
+      TR::Float, /* .dataType */ \
+      ILTypeProp::Size_4 | ILTypeProp::Floating_Point, /* .typeProperties */ \
+      ILChildProp::NoChildren, /* .childProperties */ \
+      TR::BadILOp, /* .swapChildrenOpCode */ \
+      TR::BadILOp, /* .reverseBranchOpCode */ \
+      TR::BadILOp, /* .booleanCompareOpCode */ \
+      TR::BadILOp, /* .ifCompareOpCode */ \
+      fconst, /* load float constant (32-bit ieee fp) */ \
+   ) \
+   MACRO(\
+      TR::dconst, /* .opcode */ \
+      "dconst", /* .name */ \
+      ILProp1::LoadConst, /* .properties1 */ \
+      ILProp2::ValueNumberShare, /* .properties2 */ \
+      ILProp3::LikeUse, /* .properties3 */ \
+      0, /* .properties4 */ \
+      TR::Double, /* .dataType */ \
+      ILTypeProp::Size_8 | ILTypeProp::Floating_Point, /* .typeProperties */ \
+      ILChildProp::NoChildren, /* .childProperties */ \
+      TR::BadILOp, /* .swapChildrenOpCode */ \
+      TR::BadILOp, /* .reverseBranchOpCode */ \
+      TR::BadILOp, /* .booleanCompareOpCode */ \
+      TR::BadILOp, /* .ifCompareOpCode */ \
+      dconst, /* load double constant (64-bit ieee fp) */ \
+   ) \
+   MACRO(\
+      TR::bconst, /* .opcode */ \
+      "bconst", /* .name */ \
+      ILProp1::LoadConst, /* .properties1 */ \
+      ILProp2::ValueNumberShare, /* .properties2 */ \
+      ILProp3::LikeUse, /* .properties3 */ \
+      0, /* .properties4 */ \
+      TR::Int8, /* .dataType */ \
+      ILTypeProp::Size_1 | ILTypeProp::Integer, /* .typeProperties */ \
+      ILChildProp::NoChildren, /* .childProperties */ \
+      TR::BadILOp, /* .swapChildrenOpCode */ \
+      TR::BadILOp, /* .reverseBranchOpCode */ \
+      TR::BadILOp, /* .booleanCompareOpCode */ \
+      TR::BadILOp, /* .ifCompareOpCode */ \
+      bconst, /* load byte integer constant (8-bit signed 2's complement) */ \
+   ) \
+   MACRO(\
+      TR::sconst, /* .opcode */ \
+      "sconst", /* .name */ \
+      ILProp1::LoadConst, /* .properties1 */ \
+      ILProp2::ValueNumberShare, /* .properties2 */ \
+      ILProp3::LikeUse, /* .properties3 */ \
+      0, /* .properties4 */ \
+      TR::Int16, /* .dataType */ \
+      ILTypeProp::Size_2 | ILTypeProp::Integer, /* .typeProperties */ \
+      ILChildProp::NoChildren, /* .childProperties */ \
+      TR::BadILOp, /* .swapChildrenOpCode */ \
+      TR::BadILOp, /* .reverseBranchOpCode */ \
+      TR::BadILOp, /* .booleanCompareOpCode */ \
+      TR::BadILOp, /* .ifCompareOpCode */ \
+      sconst, /* load short integer constant (16-bit signed 2's complement) */ \
+   ) \
```

Comment:
Looking better. Perhaps having the comments on the left side (since they are fixed width and consistent from one opcode to the next) may make things a lot easier to read and parse :

```
   MACRO(\
      /* .opcode               = */ TR::sconst, \
      /* .name                 = */ "sconst", \
      /* .properties1          = */ ILProp1::LoadConst, \
      /* .properties2          = */ ILProp2::ValueNumberShare,  \
      /* .properties3          = */ ILProp3::LikeUse, \
      /* .properties4          = */ 0,  \
      /* .dataType             = */ TR::Int16,  \
      /* .typeProperties       = */ ILTypeProp::Size_2 | ILTypeProp::Integer, \
      /* .childProperties      = */ ILChildProp::NoChildren, \
      /* .swapChildrenOpCode   = */ TR::BadILOp, \
      /* .reverseBranchOpCode  = */ TR::BadILOp, \
      /* .booleanCompareOpCode = */ TR::BadILOp, \
      /* .ifCompareOpCode      = */ TR::BadILOp, \
      /* .description          = */ sconst, \
   ) \
```

Reply:
That's what I thought of doing initially. However, some opcodes' last property (enum values from *OMRILOpcodesEnum.hpp*) has a long single-line comment.

Should I disregard them, or have them something like that:

```
      /* .description          = */ sconst, \ // load short integer constant (16-bit signed 2's complement)
```

Reply:
I don't see that last field in [1] at all. I suggest if there is a comment like the in your previous post you can just include it in the `.description` comment, for example:

```
...
      /* .reverseBranchOpCode  = */ TR::BadILOp, \
      /* .booleanCompareOpCode = */ TR::BadILOp, \
      /* .ifCompareOpCode      = */ TR::BadILOp, \
      /* .description          =    load short integer constant (16-bit signed 2's complement), */ \
```

[1] https://github.com/eclipse/omr/blob/1862ec43d12a8f5d216554e78a591a40726f32e0/compiler/il/OMRILOps.hpp#L42-L63

Reply:
Oops, I did not mean properties, but rather the last argument of these macros. You are right, properties end with `.ifCompareOpCode`, and then we have the opcode enum value.

Would it still be a good solution given that these comments can go up to 100 characters? E.g., [this opcode from OMRILOpCodesEnum.hpp](https://github.com/eclipse/omr/blob/master/compiler/il/OMRILOpCodesEnum.hpp#L700)

Reply:
Yeah, leave them in there for now so we preserve the information. These comments likely belong elsewhere, probably in a per IL documentation with some structure, but we're not there yet. Let's preserve them for now and once we find a good home for them we will eventually remove the comments.

Reply:
I think it is a bit better now. What do you think?

```c++
MACRO(\
   /* .opcode               = */ TR::variableNewArray, \
   /* .name                 = */ "variableNewArray", \
   /* .properties1          = */ ILProp1::HasSymbolRef, \
   /* .properties2          = */ ILProp2::CanRaiseException| ILProp2::MayUseSystemStack, \
   /* .properties3          = */ ILProp3::LikeDef, \
   /* .properties4          = */ 0, \
   /* .dataType             = */ TR::Address, \
   /* .typeProperties       = */ ILTypeProp::Reference, \
   /* .childProperties      = */ ONE_CHILD(ILChildProp::UnspecifiedChildType), \
   /* .swapChildrenOpCode   = */ TR::BadILOp, \
   /* .reverseBranchOpCode  = */ TR::BadILOp, \
   /* .booleanCompareOpCode = */ TR::BadILOp, \
   /* .ifCompareOpCode      = */ TR::BadILOp, \
   /* .description          = */ variableNewArray, /* new array - type not known at compile time, type must be a j9class, do not use type enums */ \
)
```

Reply:
Looks much better!